### DISCUSSION TITLE
How to query complex list

### DISCUSSION THREAD

**Discussion by carvalho2707** [GitLiveApp/firebase-kotlin-sdk#574] [Q&A]
With firestore how can query a collection to find if a property( list of objects) has one where a property matches a given string?

**Comment by Daeda88** [comment]
The library supports all queries documented in [here](https://firebase.google.com/docs/firestore/query-data/queries), with the same kind of restrictions. Syntax is a bit different to make it more kotlin like. From your description it's not entirely clear what you'd want to query, but generally its:

```kotlin
val collectionReference: CollectionReference = // Get a reference to your collection
collectionReference.where {
    NAME_OF_PROPERTY equalTo DESIRED_VALUE
}.get()
```
